src/linker/pipeline.py

```diff
@@ -16,81 +15,40 @@ class Pipeline:
+            ):
+                return False
+        return True
+
     #################
     # Setup methods #
     #################
 
-    def _get_steps(self) -> Tuple:
-        spec_steps = tuple(self.config.pipeline["steps"])
-        steps = tuple([x for x in spec_steps if x in STEPS])
-        unknown_steps = [x for x in spec_steps if x not in STEPS]
-        if unknown_steps:
-            logger.warning(
-                f"Unknown steps are included in the pipeline specification: {unknown_steps}.\n"
-                "These steps will be ignored.\n"
-                f"Supported steps: {STEPS}"
-            )
-        if not steps:
-            raise RuntimeError(
-                "No supported steps found in pipeline specification.\n"
-                f"Steps found in pipeline specification: {spec_steps}\n"
-                f"Supported steps: {STEPS}"
-            )
-
-        return steps
+    def _get_steps(self) -> Tuple[Step, ...]:
+        return tuple(Step(step) for step in self.config.pipeline["steps"])
 
-    def _load_implementation_metadata(self) -> Dict[str, Path]:
-        implementation_metadata = {}
+    def _get_implementations(self) -> Tuple[Implementation, ...]:
+        implementations = []
         for step in self.steps:
-            implementation_dir = self._get_implementation_directory(step)
-            metadata_path = implementation_dir / "metadata.yaml"
-            if metadata_path.exists():
-                metadata = load_yaml(metadata_path)
-                implementation_metadata[step] = metadata
-            else:
-                implementation_metadata[step] = None
-        return implementation_metadata
-
-    ##################
-    # Helper methods #
-    ##################
-
-    def _get_implementation_directory(self, step_name: str) -> Path:
-        # TODO: move this into proper config validator
-        implementation = self.config.pipeline["steps"][step_name]["implementation"]
-        implementation_dir = (
-            Path(os.path.realpath(__file__)).parent / "steps" / step_name / "implementations"
-        )
-        implementation_names = [
-            str(d.name) for d in implementation_dir.iterdir() if d.is_dir()
-        ]
-        if implementation in implementation_names:
-            implementation_dir = implementation_dir / implementation
-        else:
-            raise NotImplementedError(
-                f"No support found for step '{step_name}', implementation '{implementation}'."
-            )
-        return implementation_dir
-
-    def _get_container_full_stem(self, step_name: str) -> str:
-        container_dict = self.implementation_metadata[step_name]["image"]
-        return f"{container_dict['directory']}/{container_dict['filename']}"
+            implementation_name = self.config.pipeline["steps"][step.name]["implementation"]
+            implementations.append(Implementation(step.name, implementation_name))
```

Comment:
Nice. I can see how this would be extended to map config -> implementations when the config gets more complicated (e.g. looping a step).